agent/qrexec-agent.c

```diff
@@ -231,23 +241,23 @@ void do_exec(char *cmd)
     retval = pam_putenv(pamh, env_buf);
     if (retval != PAM_SUCCESS)
         goto error;
+    if (setgid(pw->pw_gid))
+        exit(126);
+    if (setuid(pw->pw_uid))
+        exit(126);
```

Comment:
Can you explain why you do this in the parent process? The parent process is responsible for `pam_close_session()` call, which may need privileged access (depending on PAM configuration).